File: app/src/main/java/com/adityasubathu/vaccinationreminder/AlarmReceiver.java
package com.adityasubathu.vaccinationreminder;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

public class AlarmReceiver extends BroadcastReceiver {
    @Override

    // todo: check why this isn't working
    public void onReceive(Context context, Intent intent) {
        Toast.makeText(context, "An Alarm Has Been Triggered", Toast.LENGTH_SHORT).show();

        String s = intent.getStringExtra("position");

        Intent i = new Intent(context, AlarmAlertWindow.class);
        i.putExtra("position", s);

        context.startActivity(i);

    }
}

File: app/src/main/java/com/adityasubathu/vaccinationreminder/AlarmReceiver.kt
package com.adityasubathu.vaccinationreminder

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.widget.Toast

class AlarmReceiver : BroadcastReceiver() {
    // todo: check why this isn't working
    override fun onReceive(context: Context, intent: Intent) {
        Toast.makeText(context, "An Alarm Has Been Triggered", Toast.LENGTH_SHORT).show()
        val s = intent.getStringExtra("position")
        val i = Intent(context, AlarmAlertWindow::class.java)
        i.putExtra("position", s)
        context.startActivity(i)
    }
}